modules/codegen/src/smithy4s/codegen/SmithyToIR.scala

```diff
@@ -287,18 +288,30 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
           .asScala
           .zipWithIndex
           .map { case ((name, value), index) =>
-            EnumValue(value, index, name)
+            val member = shape.getMember(name).get()
+
+            // random note: any other traits that are allowed on enum values? cuz they don't generate hints.
+            // known: @deprecated, @documentation
```

Comment:
IMO @documentation is reason enough to support this. We'd have to change the runtime enum value model, I assume.

Reply:
Good catch. The runtime enum value model [does support hints](https://github.com/disneystreaming/smithy4s/blob/main/modules/core/src/smithy4s/schema/EnumValue.scala#L25) 😉

Reply:
`EnumValue` does, true. I thought it didn't because the generated enum classes don't:

```scala
sealed abstract class FaceCard(_value: String, _name: String, _intValue: Int)

case object JACK extends FaceCard("JACK", "JACK", 1)
```

so we'll just have to change the codegen of these. If it doesn't turn out to be a can of worms I'll do that here

Reply:
f8331897f3495a92440f6d7edd3d53cea4483788 + adding it to Dynamic e18ba3a8f944a70d4d9ecbe1030b26cfe8134066